prime-router/src/main/kotlin/history/db/DeliveryDatabaseAccess.kt

```diff
@@ -167,76 +167,63 @@ class DeliveryDatabaseAccess(val db: DatabaseAccess = BaseEngine.databaseAccessS
-                    ReportFile.REPORT_FILE.REPORT_ID.`as`(DeliveryTable.DELIVERY.REPORT_ID),
-                    ReportFile.REPORT_FILE.CREATED_AT.`as`(DeliveryTable.DELIVERY.CREATED_AT),
-                    // Currently an open issue for doing this via the DSL
-                    // https://github.com/jOOQ/jOOQ/issues/6723
-                    DSL.field(
-                        "\"public\".\"report_file\".\"created_at\" + INTERVAL '$EXPIRATION_DAYS_OFFSET days'",
-                        SQLDataType.OFFSETDATETIME
-                    )
-                        .`as`(DeliveryTable.DELIVERY.EXPIRATION_DATE),
-                    DSL.sum(ReportFile.REPORT_FILE.ITEM_COUNT).`as`(DeliveryTable.DELIVERY.TEST_RESULT_COUNT),
-                    ReportFile.REPORT_FILE.REPORT_ID.cast(SQLDataType.VARCHAR)
-                        .concat(
-                            metadata.field(
-                                CovidResultMetadata
-                                    .COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
-                            )
-                        )
-                        .concat(
-                            metadata.field(
-                                CovidResultMetadata
-                                    .COVID_RESULT_METADATA.ORDERING_FACILITY_NAME
-                            )
-                        )
-                        .concat(metadata.field(CovidResultMetadata.COVID_RESULT_METADATA.SENDER_ID))
-                        .`as`(DeliveryTable.DELIVERY.SORT_ID)
-                )
-                    .from(metadata)
-                    .join(ItemGraphTable.ITEM_GRAPH)
-                    .on(
-                        ItemGraphTable.ITEM_GRAPH.PARENT_REPORT_ID
-                            .eq(metadata.field(CovidResultMetadata.COVID_RESULT_METADATA.REPORT_ID)),
-                        ItemGraphTable.ITEM_GRAPH.PARENT_INDEX
-                            .eq(metadata.field(CovidResultMetadata.COVID_RESULT_METADATA.REPORT_INDEX))
                     )
-                    .join(ReportFile.REPORT_FILE)
-                    .on(ReportFile.REPORT_FILE.REPORT_ID.eq(ItemGraphTable.ITEM_GRAPH.STARTING_REPORT_ID))
-                    .groupBy(
-                        ReportFile.REPORT_FILE.REPORT_ID,
-                        metadata.field(
-                            CovidResultMetadata
-                                .COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
-                        ),
-                        metadata.field(CovidResultMetadata.COVID_RESULT_METADATA.ORDERING_FACILITY_NAME),
-                        metadata.field(CovidResultMetadata.COVID_RESULT_METADATA.SENDER_ID),
-                    ).coerce(DeliveryTable.DELIVERY)
+                    .concat(CovidResultMetadata.COVID_RESULT_METADATA.SENDER_ID)
+                    .`as`(DeliveryTable.DELIVERY.SORT_ID)
             )
+                .from(CovidResultMetadata.COVID_RESULT_METADATA)
+                .join(ItemGraphTable.ITEM_GRAPH)
```

Comment:
The primary change here is switching to joining against the item graph directly rather than going through a CTE.